### PR TITLE
eo metrics for Telemetry

### PR DESCRIPTION
**Pull request by pmoogi-redhat**
This PR contains whitelisting of the elasticsearch-operator metrics for Telemetry captured in the below document:
  mentioned in this doc https://docs.google.com/document/d/1ek98QDTztWHuYzUX_T53O9JblUtL9-YRMUK5oHMIK8w/edit

- Changes done in manifests/0000_50_cluster-monitoring-operator_04-config.yaml to include the above metrics.
- Reference https://issues.redhat.com/browse/LOG-2095 andhttps://issues.redhat.com/browse/LOG-2096 [JIRA ticket for implementing EO Metrics]
- Reference https://issues.redhat.com/browse/LOG-1852 [for selecting EO metrics for Telemetry]
- https://github.com/openshift/elasticsearch-operator/blob/master/internal/metrics/metrics.go [EO Metrics implementation]

### CODE REVIEW

**Comment by pmoogi-redhat** [comment]
@jan--f kindly review this PR.

**Comment by jan--f** [comment]
/lgtm
/label docs-approved
cc @sferich888 for px approval
cc @juzhao @lihongyan1 for qe approval
@pmoogi-redhat please note that telemeter ingestion will require allow-listing on the telemeter side as well as detailed in point 6 in https://docs.google.com/document/d/1a6n5iBGM2QaIQRg9Lw4-Npj6QY9--Hpx3XYut-BrUSY/edit#heading=h.i05nm4wmvn7x

**Review by PhilipGough — APPROVED** [review]

File: Documentation/data-collection.md [hunk 1]
```diff
@@ -626,6 +626,66 @@ data:
...
+    # owners: (@openshift/team-logging)
+    #
+    # Number of redundancy policies used across the fleet.
+    - '{__name__="eo_es_redundancy_policy_info"}'
+    #
+    # elasticsearch operator metrics for Telemetry
+    # owners: (@openshift/team-logging)
+    #
+    # Max/min document age used per policy across the fleet.
+    - '{__name__="cluster:eo_es_documents_created_total:sum"}'
```
Comment (juzhao) [inline Documentation/data-collection.md:646]: two `cluster:eo_es_documents_created_total:sum` in this file

File: Documentation/data-collection.md [hunk 2]
```diff
@@ -626,6 +626,66 @@ data:
...
+    # elasticsearch operator metrics for Telemetry
+    # owners: (@openshift/team-logging)
+    #
+    # Number of namespaces deleted per policy across the fleet.
+    - '{__name__="eo_es_defined_delete_namespaces_total"}'
+    #
+    # elasticsearch operator metrics for Telemetry
+    # owners: (@openshift/team-logging)
+    #
+    # Number of clusters with misconfigured cpu or memory resources across the fleet.
```
Comment (juzhao) [inline Documentation/data-collection.md:657]: it seems the metric is only for memory, does not include cpu

File: Documentation/sample-metrics.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ return the full set of metrics that the Telemeter client captures:
...
...e_vcenter_info:sum|cluster:vsphere_esxi_version_total:sum|cluster:vsphere_node_hw_version_total:sum|openshift:build_by_strategy:sum|rhods_aggregate_availability|rhods_total_users|instance:etcd_disk_wal_fsync_duration_seconds:histogram_quantile|instance:etcd_mvcc_db_total_size_in_bytes:sum|instance:etcd_network_peer_round_trip_time_seconds:histogram_quantile|instance:etcd_mvcc_db_total_size_in_use_in_bytes:sum|instance:etcd_disk_backend_commit_duration_seconds:histogram_quantile|jaeger_operator_instances_storage_types|jaeger_operator_instances_strategies|jaeger_operator_instances_agent_strategies|appsvcs:cores_by_product:sum|nto_custom_profiles:count|openshift_csi_share_configmap|openshift_csi_share_secret|openshift_csi_share_mount_failures_total|openshift_csi_share_mount_requests_total|eo_es_storage_info|eo_es_redundancy_policy_info|cluster:eo_es_documents_created_total:sum|eo_es_defined_delete_namespaces_total|eo_es_misconfigured_memory_resources_info|cluster:eo_es_data_nodes_total:max|cluster:eo_es_documents_created_total:sum|cluster:eo_es_documents_deleted_total:sum|pod:eo_es_shards_total:max|eo_es_cluster_management_state_info",alertstate=~"firing|",quantile=~"0.99|0.99|0.99|"}
```
Comment (juzhao) [inline Documentation/sample-metrics.md:16]: two `cluster:eo_es_documents_created_total:sum` in this file

File: Documentation/telemetry/telemeter_query [hunk 4]
```diff
@@ -1 +1 @@
...
...e_vcenter_info:sum|cluster:vsphere_esxi_version_total:sum|cluster:vsphere_node_hw_version_total:sum|openshift:build_by_strategy:sum|rhods_aggregate_availability|rhods_total_users|instance:etcd_disk_wal_fsync_duration_seconds:histogram_quantile|instance:etcd_mvcc_db_total_size_in_bytes:sum|instance:etcd_network_peer_round_trip_time_seconds:histogram_quantile|instance:etcd_mvcc_db_total_size_in_use_in_bytes:sum|instance:etcd_disk_backend_commit_duration_seconds:histogram_quantile|jaeger_operator_instances_storage_types|jaeger_operator_instances_strategies|jaeger_operator_instances_agent_strategies|appsvcs:cores_by_product:sum|nto_custom_profiles:count|openshift_csi_share_configmap|openshift_csi_share_secret|openshift_csi_share_mount_failures_total|openshift_csi_share_mount_requests_total|eo_es_storage_info|eo_es_redundancy_policy_info|cluster:eo_es_documents_created_total:sum|eo_es_defined_delete_namespaces_total|eo_es_misconfigured_memory_resources_info|cluster:eo_es_data_nodes_total:max|cluster:eo_es_documents_created_total:sum|cluster:eo_es_documents_deleted_total:sum|pod:eo_es_shards_total:max|eo_es_cluster_management_state_info",alertstate=~"firing|",quantile=~"0.99|0.99|0.99|"}
```
Comment (juzhao) [inline Documentation/telemetry/telemeter_query:1]: two `cluster:eo_es_documents_created_total:sum` in this file

**Comment by jan--f** [comment]
/hold
@pmoogi-redhat please address @juzhao review comments.

**Comment by pmoogi-redhat** [comment]
@jan--f @juzhao addressed a duplicate metric issue in the whitelisting metrics manifest.

**Comment by pmoogi-redhat** [comment]
@juzhao  addressed your review comments. kindly check. thank you.

File: Documentation/data-collection.md [hunk 5]
```diff
@@ -648,6 +648,60 @@ data:
...
+    - '{__name__="eo_es_misconfigured_memory_resources_info"}'
```
Comment (juzhao) [inline Documentation/data-collection.md:674]: it seems the metric is only for memory, does not include cpu

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 6]
```diff
@@ -640,6 +640,60 @@ data:
...
+    # owners: (@openshift/team-logging)
+    #
+    # Number of namespaces deleted per policy across the fleet.
+    - '{__name__="eo_es_defined_delete_namespaces_total"}'
+    #
+    # elasticsearch operator metrics for Telemetry
+    # owners: (@openshift/team-logging)
+    #
+    # Number of clusters with misconfigured cpu or memory resources across the fleet.
+    - '{__name__="eo_es_misconfigured_memory_resources_info"}'
```
Comment (juzhao) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:666]: it seems the metric is only for memory, does not include cpu

**Comment by juzhao** [comment]
just one comment, others are LGTM

File: manifests/0000_50_cluster-monitoring-operator_04-config.yaml [hunk 7]
```diff
@@ -640,6 +640,60 @@ data:
...
+    # elasticsearch operator metrics for Telemetry
+    # owners: (@openshift/team-logging)
+    #
+    # Number of namespaces deleted per policy across the fleet.
+    - '{__name__="eo_es_defined_delete_namespaces_total"}'
+    #
+    # elasticsearch operator metrics for Telemetry
+    # owners: (@openshift/team-logging)
+    #
+    # Number of clusters with misconfigured cpu or memory resources across the fleet.
```
Comment (jan--f) [inline manifests/0000_50_cluster-monitoring-operator_04-config.yaml:665]:
```suggestion
    # Number of clusters with misconfigured memory resources across the fleet.
```

**Review by jan--f — CHANGES_REQUESTED** [review]

File: Documentation/data-collection.md (hunk 5 above)
Comment (juzhao) [reply Documentation/data-collection.md:674]: @jan--f this should also be updated

**Comment by jan--f** [comment]
/lgtm
/label docs-approved
/label px-approved
Adding labels because a telemetry addition has no user or doc impact.
Ping @juzhao for qe-approved.

**Comment by jan--f** [comment]
/lgtm cancel
Sorry I failed to notice that the generate job still fails.

**Pull request merged**